samanthas_telegram_bot/conversation/callbacks/teacher.py

```diff
@@ -0,0 +1,214 @@
+        return ConversationState.ASK_TEACHING_FREQUENCY
+
+    # inexperienced teacher only get one group
+    context.user_data.teacher_number_of_groups = 1
+    await CQReplySender.ask_teaching_frequency(context, query)
+    return ConversationState.PREFERRED_STUDENT_AGE_GROUPS_START
+
+
+async def store_number_of_groups_ask_frequency(
+    update: Update, context: CUSTOM_CONTEXT_TYPES
+) -> int:
+    """For experienced teachers: stores information about number of groups, asks for frequency
+    (inexperienced teachers).
+    """
+    query, context.user_data.teacher_number_of_groups = await answer_callback_query_and_get_data(
+        update
+    )
+
+    await CQReplySender.ask_teaching_frequency(context, query)
+    return ConversationState.PREFERRED_STUDENT_AGE_GROUPS_START
+
+
+async def store_frequency_ask_student_age_groups(
+    update: Update, context: CUSTOM_CONTEXT_TYPES
+) -> int:
+    """Stores frequency, asks for preferred age groups of students."""
+    query, context.user_data.teacher_class_frequency = await answer_callback_query_and_get_data(
+        update
+    )
+
+    await CQReplySender.ask_teacher_age_groups_of_students(context, query)
+
+    return ConversationState.PREFERRED_STUDENT_AGE_GROUPS_MENU_OR_ASK_NON_TEACHING_HELP
+
+
+async def store_student_age_group_ask_another_or_non_teaching_help(
+    update: Update, context: CUSTOM_CONTEXT_TYPES
+) -> int:
+    """Stores preferred age group of students, asks another.  If the teacher is done, ask about
+    additional help for students.
+    """
+    query, data = await answer_callback_query_and_get_data(update)
+
+    if data != CommonCallbackData.DONE:
+        context.user_data.teacher_age_groups_of_students.append(data)
+
+    # teacher pressed "Done" or chose all age groups
+    if data == CommonCallbackData.DONE or len(
+        context.user_data.teacher_age_groups_of_students
+    ) == len(STUDENT_AGE_GROUPS_FOR_TEACHER):
+        await CQReplySender.ask_non_teaching_help(context, query)
+        return (
+            ConversationState.NON_TEACHING_HELP_MENU_OR_PEER_HELP_FOR_TEACHER_OR_REVIEW_FOR_STUDENT
+        )
+
+    await CQReplySender.ask_teacher_age_groups_of_students(context, query)
+    return ConversationState.PREFERRED_STUDENT_AGE_GROUPS_MENU_OR_ASK_NON_TEACHING_HELP
+
+
+async def store_peer_help_ask_another_or_additional_help(
+    update: Update, context: CUSTOM_CONTEXT_TYPES
+) -> int:
+    """Stores one option of teacher peer help, asks for another.  If the teacher is done,
+    asks for any additional help (in free text)."""
+    query, data = await answer_callback_query_and_get_data(update)
+
+    if data == CommonCallbackData.DONE:
+        await query.edit_message_text(
+            PHRASES["ask_teacher_any_additional_help"][context.user_data.locale],
+            reply_markup=InlineKeyboardMarkup([]),
+        )
+        return ConversationState.ASK_REVIEW
+
+    if data == "consult":
```

Comment:
Maybe some dict here with map from help_type to field of teacher_peer_help?

Reply:
That's a good thought. Although I've been thinking that maybe I should scrap this altogether and replace this bunch of boolean flags with a list (or a set) in the fashion similar to [non_teaching_help](https://github.com/samanthasgroup/samanthas-telegram-bot/blob/students-bot-1st-version/samanthas_telegram_bot/conversation/user_data.py#L41). All these literals depend on ID's of phrases in `phrases.csv` anyway.

Reply:
So let's keep a TODO here? :)

Reply:
I can keep a TODO for now. Do you see any advantages of an iterable (like in non_teaching_help) or a bunch of boolean flags?

Reply:
OK, I'll replace it with a list or set in the next PR :)